src/client/public/scripts/dash/dash.ts

```diff
@@ -151,6 +169,57 @@ function onMessage(event: ServiceWorkerMessageEvent|MessageEvent) {
 }
 
 async function start() {
+  // Render persistent UI.
+  const outgoingFilters: LegendItem[] = [
+    {type: 'complete', description: 'Ready to merge'},
+    {type: 'actionable', description: 'Requires attention'},
+    {type: 'activity', description: 'New activity'},
+  ];
+  const incomingFilters: LegendItem[] = [
+    {type: 'actionable', description: 'Requires attention'},
+    {type: 'activity', description: 'New activity'},
+  ];
+  const assignedFilters: LegendItem[] = [
+    {type: 'actionable', description: 'Assigned to you'},
+  ];
+
+  render(
+      legendTemplate(outgoingFilters),
+      document.querySelector('.outgoing-legend') as HTMLElement);
+  render(
+      legendTemplate(incomingFilters),
+      document.querySelector('.incoming-legend') as HTMLElement);
+  render(
+      legendTemplate(assignedFilters),
+      document.querySelector('.assigned-issues-legend') as HTMLElement);
+
+  filterController.createFilter('outgoing-prs', outgoingFilters);
+  filterController.createFilter('incoming-prs', outgoingFilters);
+  filterController.createFilter('assigned-issues', assignedFilters);
+
+  /**
+   * Event handler for when the filter is changed.
+   */
+  function filterChanged(id: FilterId, event: {detail: FilterState}) {
+    filterController.updateFilter(id, event.detail);
+    renderAll();
+  }
+
+  /**
+   * Finds and attaches filter event listener.
+   */
+  function attachFilterListener(id: FilterId) {
+    const element = document.getElementById(id);
+    if (!element) {
+      throw Error('Could not attach filter event listener');
+    }
+    element.addEventListener('legend-change', filterChanged.bind(null, id));
```

Comment:
You could dispatch an event on the window and use a data field to determine behavior - It's how the automerge behavior is currently working.